### ISSUE TITLE
doctest skipif may remove the code block from documentation

### ISSUE THREAD

**Issue by wojdyr** [sphinx-doc/sphinx#6068]
Doctest `:skipif:` option (introduced in #5307) may unintentionally remove the code in doctest from documentation.
Ideally, the `:skipif:` condition would be evaluated only by the doctest builder.

**Comment by tk0miya** [comment]
Could you give any example?

**Comment by wojdyr** [comment]
contents.rst:
```py
.. doctest::

  >>> 1
  1

.. doctest::
  :skipif: True

  >>> 2
  2

.. doctest::
  :skipif: False

  >>> 3
  3
```

Tested as follows:

```sh
$ sphinx-build -b html -C -D extensions=sphinx.ext.doctest . out/
$ lynx -dump out/contents.html 
   #[1]Index [2]Search

>>> 1
1

>>> 3
3

...
```

It is missing:
```
  >>> 2
  2
```
in the output.

**Comment by tk0miya** [comment]
Thank you for example. confirmed. Surely, this is not intentional.

**Comment by tk0miya** [comment]
Fixed by #6115.
Thank you for reporting!
